src/t_cose_sign1_verify.c

```diff
@@ -308,51 +461,32 @@ t_cose_sign1_verify_internal(struct t_cose_sign1_verify_ctx *me,
         goto Done;
     }
 
-
-    /* -- Skip signature verification if requested --*/
-    if(me->option_flags & T_COSE_OPT_DECODE_ONLY) {
-        return_value = T_COSE_SUCCESS;
-        goto Done;
-    }
-
-
-    /* -- Compute the TBS bytes -- */
-    return_value = create_tbs_hash(parameters.cose_algorithm_id,
-                                   protected_parameters,
-                                   aad,
-                                   *payload,
-                                   buffer_for_tbs_hash,
-                                   &tbs_hash);
-    if(return_value) {
-        goto Done;
-    }
-
-
-    /* -- Check for short-circuit signature and verify if it exists -- */
 #ifndef T_COSE_DISABLE_SHORT_CIRCUIT_SIGN
     short_circuit_kid = get_short_circuit_kid();
     if(!q_useful_buf_compare(parameters.kid, short_circuit_kid)) {
-        if(!(me->option_flags & T_COSE_OPT_ALLOW_SHORT_CIRCUIT)) {
-            return_value = T_COSE_ERR_SHORT_CIRCUIT_SIG;
-            goto Done;
-        }
-
-        return_value = t_cose_crypto_short_circuit_verify(tbs_hash, signature);
+        return_value = sign1_verify_short_circuit(me, &parameters, signature, protected_parameters, aad, signed_payload);
         goto Done;
     }
 #endif /* T_COSE_DISABLE_SHORT_CIRCUIT_SIGN */
 
+#ifndef T_COSE_DISABLE_EDDSA
+    if (parameters.cose_algorithm_id == COSE_ALGORITHM_EDDSA) {
+        return_value = sign1_verify_eddsa(me, &parameters, signature, protected_parameters, aad, signed_payload);
+        goto Done;
+    }
+#endif
 
-    /* -- Verify the signature (if it wasn't short-circuit) -- */
-    return_value = t_cose_crypto_verify(parameters.cose_algorithm_id,
-                                        me->verification_key,
-                                        parameters.kid,
-                                        tbs_hash,
-                                        signature);
+    return_value = sign1_verify_default(me, &parameters, signature, protected_parameters, aad, signed_payload);
 
 Done:
-    if(returned_parameters != NULL) {
-        *returned_parameters = parameters;
+    if (return_value == T_COSE_SUCCESS)
+    {
+        if(returned_parameters != NULL) {
+            *returned_parameters = parameters;
+        }
+        if(!is_dc && payload != NULL) {
+            *payload = signed_payload;
```

Comment:
signed_payload is uninitialized here when header_decode fails above

Reply:
This line only executes if return_value == T_COSE_SUCCESS, so everything should be initialized correctly.